### DISCUSSION TITLE
Custom fields native exporter and importer?

### DISCUSSION THREAD

**Discussion by giudansky** [joomla/joomla-cms#47063] [2. Cold Features]
Custom fields are probably the best joomla feature so far. I'm using it a lot and it's very interconnected with my templates developing.
Now... I was wondering why it seems there are no ways to batch reproduce custom fields. A template using custom fields needs to recreate every single field on every joomla installation.
It would be so nice having an xml inside the template to reconfigure them on the fly when installing the template. Or an import/export procedure in the custom fields dashboard.
If you have 10 custom fields each one with options, conditionals and field types it means recreating them one by one and it's frustrating. I know I can import/export with an sql query but thats not exactly accessible.

Thanks

**Comment by jduerscheid** [comment]
This is in my opionion an good feature, but not really easy to do, because you there are a few things to think about. 

- what happens to the existing fields during the import
- not only the fields had to be exported and imported, also the field groups
- how to manage if ID's are changed during the import for subforms and conditional fields

**Comment by brianteeman** [comment]
This extension should help you https://github.com/pjdevries/com_hopper

**Comment by giudansky** [comment]
Thanks I'll check it out!
